src/Extensions/WorkflowEmbargoExpiryExtension.php

```diff
@@ -195,7 +195,7 @@ public function clearUnPublishJob()
     {
         // Cancel any in-progress unpublish job
```

Comment:
This comment seems to indicate that we're meant to explicitly be deleting jobs that aren't finished? So presumably the other method is also doing that intentionally?
I haven't looked any more deeply than that, I was just having a quick look to answer the question of which branch to target

Reply:
Thanks for taking the time @GuySartorelli.

Can you maybe rephrase your comment? I'm not sure I fully get which other method you mean and what you think should or shouldn't be happening.

Besides that, I think there's definitely room for improvement around the jobs handling within the module here, this patch is a step forward but a proper review of what jobs are created, executed, removed, updated etc. would still be beneficial. I reckon there were enhancements in the queued jobs module over time that could be leveraged here (just a gut feeling, no specifics in mind).

Reply:
Sorry yeah.

This comment is in `clearUnPublishJob()` - `clearPublishJob()` is very similar and is the other method that was altered in this pr, though it lacks a comment saying anything about jobs that are still running. I'm guessing they're intended to function the same way as one another, so if this method is meant to delete unfinished jobs, the publish method probably is as well.

I'm not sure what should or shouldn't be happening - I just noticed that this comment in the code seems to be contrary to the change being made. I haven't looked at what this functionality is doing, or when it is calle,d or what the side effects ,ight be etc as that wasn't part of the type of review I was doing. I'll leave it to you and @wilr to make determinations as to what should be happening here.

Reply:
Yep, those methods should behave the same way, the other is just missing the comment. The publish/unpublish are just two variants of the same action and the jobs should behave the same. I'd say the other method just doesn't have the comment (rather than being different in principle).